src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -156,33 +125,32 @@ public void Running_Timer_CanBeFinalizedAndStopsFiring()
     {
         AutoResetEvent are = new AutoResetEvent(false);
         TimerCallback tc = new TimerCallback((object o) => are.Set());
-        var t = new Timer(tc, null, 100, 500);
-        Assert.True(are.WaitOne(250), "Failed to get first timer fire");
+        var t = new Timer(tc, null, 1, 500);
+        Assert.True(are.WaitOne(MaxPositiveTimeoutInMs), "Failed to get first timer fire");
         t = null; // Remove our refence so the timer can be GC'd 
         GC.Collect();
         GC.WaitForPendingFinalizers();
         GC.Collect();
-        Assert.False(are.WaitOne(750), "Should not have received a timer fire after it was collected");
+        Assert.False(are.WaitOne(500), "Should not have received a timer fire after it was collected");
     }
 
     [Fact]
-    public void MultpleTimers_PeriodicTimerIsntBlockedByABlockedTimer()
+    public void MultpleTimers_PeriodicTimerIsntBlockedByBlockedCallback()
     {
-        AutoResetEvent are1 = new AutoResetEvent(false);
-        AutoResetEvent are2 = new AutoResetEvent(false);
-        TimerCallback tc = new TimerCallback((object o) =>
+        ManualResetEventSlim mres = new ManualResetEventSlim();
+        int callbacks = 2;
+        Barrier b = new Barrier(callbacks + 1);
+        Timer t = null;
+        t = new Timer(_ =>
         {
-            using (var t1 = new Timer((object obj) => are1.Set(), null, 1, -1))
+            if (Interlocked.Decrement(ref callbacks) >= 0)
             {
-                Assert.True(are1.WaitOne(250), "Should have received a second callback while blocked");
-                are2.Set();
+                Assert.True(b.SignalAndWait(MaxPositiveTimeoutInMs));
             }
-        });
+            t.Dispose();
+        }, null, 1, 50);
```

Comment:
In my code, I had:
```C#
}, null, -1, -1);
t.Change(1, 50);
```
(though I think I wrote it like the above initially, so maybe you got an email or something that contained the old version and based this on that).  This is subtly different from what you have here:
```C#
}, null, 1, 50);
```
but in an important way.  The version that just sets the times in the ctor is subject to a race condition, where the callback could execute in another thread before the calling thread is able to assign the constructed Timer instance to ```t```, such that the callback then tries to execute ```t.Dispose()``` and gets a NullReferenceException.  By instead passing -1,-1 to the constructor and then subsequently calling t.Change(1, 50), we guarantee that ```t``` has been assigned by the time the callback is executed.